eng/validation/build-arcadewithrepo.ps1

```diff
@@ -0,0 +1,265 @@
+Param(
+  [Parameter(Mandatory=$true)]
+  [string] $azdoOrg, 
+
+  [Parameter(Mandatory=$true)]
+  [string] $azdoProject,
+
+  [Parameter(Mandatory=$true)]
+  [int] $buildDefinitionId,
+
+  [Parameter(Mandatory=$true)]
+  [string] $azdoToken,
+
+  [Parameter(Mandatory=$true)]
+  [string] $githubUser,
+  
+  [Parameter(Mandatory=$true)]
+  [string] $githubPAT,
+
+  [Parameter(Mandatory=$true)]
+  [string] $githubOrg, 
+
+  [Parameter(Mandatory=$true)]
+  [string] $targetRepoName,
+
+  [Parameter(Mandatory=$true)]
+  [string] $barToken, 
+
+  [string] $buildParameters = ''
+)
+
+set-strictmode -version 2.0
+$ErrorActionPreference = 'Stop'
+
+. $PSScriptRoot\..\common\tools.ps1
+. $PSScriptRoot\..\common\darc-init.ps1
+
+$arcadeSdkPackageName = 'Microsoft.DotNet.Arcade.Sdk'
+$arcadeSdkVersion = $GlobalJson.'msbuild-sdks'.$arcadeSdkPackageName
+
+# Get a temporary directory for a test root. Use the agent work folder if running under azdo, use the temp path if not.
+$testRootBase = if ($env:AGENT_WORKFOLDER) { $env:AGENT_WORKFOLDER } else { $([System.IO.Path]::GetTempPath()) }
+$testRoot = Join-Path -Path $testRootBase -ChildPath $([System.IO.Path]::GetRandomFileName())
+New-Item -Path $testRoot -ItemType Directory | Out-Null
+
+$days = 3
+$minTime = (Get-Date).AddDays(-$days)
+$buildReasonsList = @("batchedCI", "individualCI")
+
+function Get-LastKnownGoodBuildSha(
+    [string] $azdoOrg, 
+    [string] $azdoProject,
+    [int] $buildDefinitionId,
+    [string] $azdoToken)
+{
+
+    ## Have there been any builds in the last $days days? 
+    $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(":${azdoToken}"))
+    $headers = @{"Authorization"="Basic $base64AuthInfo"}
+    $count = 0
+
+    foreach($reason in $buildReasonsList)
+    {
+        $uri = "https://dev.azure.com/${azdoOrg}/${azdoProject}/_apis/build/builds?api-version=5.1&statusFilter=completed&definitions=${buildDefinitionId}&reasonFilter=${reason}&minTime=${minTime}&`$top=1"
+        $reponse = (Invoke-WebRequest -Uri $uri -Headers $headers -Method Get) | ConvertFrom-Json
+        $count += $reponse.count
```

Comment:
Is there a reason not to just get the build already? (as in LN95)

Reply:
Line 95 is specifically looking for successful builds in the last three days. This first check is to see if there are any completed builds (regardless of success or fail or whatever is in between). If you think the API calls are too expensive, I can get all the completed builds on 95 and then filter through them to see which are successes or non-successes.

Reply:
Right, that nuance was still confusing me lol. The API call is not expensive so it's fine to make it.